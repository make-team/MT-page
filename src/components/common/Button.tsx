import React from "react";
import styled from "styled-components";

export interface PropTypes {
  children: string;
  onClick?: () => void;
}

function Button({ children, onClick }: PropTypes) {
  return <Wrapper onClick={onClick}>{children}</Wrapper>;
}

export default Button;

const Wrapper = styled.button`
  cursor: pointer;
  transition: all 0.6s cubic-bezier(0.9, 0.24, 0.4, 1);
  color: red;
  &:hover {
    box-shadow: inset 00px 100px 0px 0px #6098ff;
    color: #ffffff;
  }
  &:hover::before {
    opacity: 1;
    height: 100%;
  }
`;
