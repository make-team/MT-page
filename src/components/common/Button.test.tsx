import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import Button from "./Button";

describe("<Button />", () => {
  it("랜더링이 됐는지", () => {
    const { container } = render(<Button onClick={() => {}}>test</Button>);

    const label = screen.getByText("test");
    expect(label).toBeInTheDocument();

    const parent = label.parentElement;

    expect(parent).toHaveStyleRule("color", "inherit");
    expect(parent).toHaveStyleRule("color", "#ffffff", {
      modifier: `:hover`,
    });
    expect(container).toMatchSnapshot();
  });
});
