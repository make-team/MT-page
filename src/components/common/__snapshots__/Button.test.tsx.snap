// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> 랜더링된 버튼의 동작확인 1`] = `
.c0 {
  cursor: pointer;
  -webkit-transition: all 0.6s cubic-bezier(0.9,0.24,0.4,1);
  transition: all 0.6s cubic-bezier(0.9,0.24,0.4,1);
}

.c0:hover {
  box-shadow: inset 00px 100px 0px 0px #6098ff;
  color: #ffffff;
}

<div>
  <button
    class="c0"
  >
    test
  </button>
</div>
`;
